packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
+          size='small'
+          aria-label='More options'
+          onClick={onButtonClick}
+        >
+          <SvgMore />
+        </IconButton>
+      </Box>
+    </DropdownMenu>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileMoreOptionsOwnProps>;
+TileMoreOptions.displayName = 'Tile.MoreOptions';
+
+// ----------------------------------------------------------------------------
+// Tile.Buttons component
+
+const TileButtons = polymorphic.div('iui-tile-buttons');
+TileButtons.displayName = 'Tile.Buttons';
+
+/**
+ * Tile with customizable Thumbnail, Name, Content and Buttons subcomponents
+ * @example
+ * <Tile>
+ *    <Tile.ThumbnailArea>
+ *      <Tile.ThumbnailPicture/>
+ *      <Tile.Badge/>
+ *      <Tile.TypeIndicator/>
+ *      <Tile.QuickAction/>
+ *    </Tile.ThumbnailArea>
+ *    <Tile.Name>
+ *      <Tile.NameIcon/>
+ *      <Tile.NameLabel/>
+ *    <Tile.Name/>
+ *    <Tile.ContentArea>
+ *      <Tile.Description />
+ *      <Tile.Metadata/>
+ *      <Tile.MoreOptions/>
+ *    </Tile.ContentArea>
+ *    <Tile.Buttons/>
+ * </Tile>
+ */
+export const Tile = Object.assign(TileComponent, {
+  /**
+   *ThumbnailArea subcomponent that contains `ThumbnailPicture`, `QuickAction`, `TypeIndicator` or `BadgeContainer`
+   * @example
+   * <Tile.ThumbnailArea>
+   *    <Tile.ThumbnailPicture/>
+   *    // or
+   *    <Tile.ThumbnailAvatar/>
+   *    <Tile.QuickAction/>
+   *    <Tile.TypeIndicator/>
+   *    <Tile.BadgeContainer/>
+   * </Tile.ThumbnailArea>
+   */
+  ThumbnailArea: TileThumbnailArea,
+  /**
+   * Thumbnail image url, a custom component or an svg for thumbnail avatar.
+   * @example
+   * <Tile>
+   *  // ...
+   *  <Tile.ThumbnailArea>
+   *    <Tile.ThumbnailPicture url = '/url/to/image.jpg'/>
+   *  </Tile.ThumbnailArea>
+   * </Tile>
+   * or
+   * <Tile>
+   *  // ...
+   *  <Tile.ThumbnailArea>
+   *    <Tile.ThumbnailPicture>
+   *      {<Avatar image={<img src='icon.png' />} />}
+   *      // or
+   *      {<SvgImodelHollow />}
+   *    </Tile.ThumbnailPicture>
+   *   </Tile.ThumbnailArea>
+   * /Tile>
+   */
+  ThumbnailPicture: TileThumbnailPicture,
+  /**
+   * `QuickAction` subcomponent shown on top left of the tile.
+   * Recommended to use an invisible `IconButton`.
+   */
+  QuickAction: TileQuickAction,
+  /**
+   * `TypeIndicator` subcomponent shown on top left of the tile.
+   * Recommended to use an invisible `IconButton`.
+   */
+  TypeIndicator: TileTypeIndicator,
+  /**
+   * `BadgeContainer` subcomponent shown on the bottom right of thumbnail.
+   */
+  BadgeContainer: TileBadgeContainer,
+  /**
+   * `IconButton` subcomponent: custom icon for `QuickAction` and `TypeIndicator` buttons.
+   */
+  IconButton: TileIconButton,
+  /**
+   * `Name` subcomponent under thumbnail or top of the Tile if no thumbnail present.
+   */
+  Name: TileName,
+  /**
+   * `NameIcon` next to name of the tile. Goes under <Tile.Name>
+   * @example
+   * <Tile>
+   * <Tile.Name>
```

Comment:
nitpicky, but the tabbing is off slightly in this example and in the `NameLabel` example

Reply:
Updated :D